src/internal/packager/sbom/viewer/compare.js

```diff
@@ -0,0 +1,173 @@
+            rightReader.readAsText(rightJson);
+
+            rightReader.onload = function() {
+                try {
+                    let rightData = JSON.parse(rightReader.result);
+                    const rightMap = {}
+                    rightData.artifacts.map(artifact => {
+                        if (!rightMap[artifact.name]) { rightMap[artifact.name] = {} }
+                        rightMap[artifact.name][artifact.version] = artifact
+                    });
+
+                    let differences = [];
+                    rightData.artifacts.map(artifact => {
+                        if (!leftMap[artifact.name]) {
+                            artifact.zarfDiff = "Added"
+                            differences.push(artifact)
+                        } else if (!leftMap[artifact.name][artifact.version]) {
+                            artifact.zarfDiff = "Changed"
+                            oldVersion = Object.keys(leftMap[artifact.name])[0]
+                            artifact.version = oldVersion + " -> " + artifact.version
+                            differences.push(artifact)
+                        }
+                    });
+
+                    leftData.artifacts.map(artifact => {
+                        if (!rightMap[artifact.name]) {
+                            artifact.zarfDiff = "Removed"
+                            differences.push(artifact)
+                        }
+                    });
+
+                    loadDataTable(differences, artifactsTable)
+                } catch (e) {
+                    showModal("Unable to Compare", "You must select 2 Syft JSON files")
+                }
+            };
+        } catch (e) {
+            showModal("Unable to Compare", "You must select 2 Syft JSON files")
+        }
+    };
+}
+
+function loadDataTable(artifacts, dataTable) {
+    const transformedData = artifacts.map(artifact => {
+        return [
+            diff(artifact.zarfDiff),
+            artifact.type,
+            artifact.name,
+            artifact.version,
+            fileList(artifact.metadata),
+            (artifact.metadata && artifact.metadata.description) || "-",
+            ((artifact.metadata && artifact.metadata.maintainer) || "-").replace(/\u003c(.*)\u003e/, `&nbsp;|&nbsp;&nbsp;<a href="mailto:$1">$1</a>`),
+            (artifact.metadata && artifact.metadata.installedSize) || "-",
+        ];
+    });
+
+    const data = {
+        "headings": [
+            "Difference",
+            "Type",
+            "Name",
+            "Version",
+            "Files",
+            "Notes",
+            "Maintainer",
+            "Size",
+        ],
+        "data": transformedData,
+    };
+
+    if (window.dt) {
+        window.dt.destroy();
+    }
+
+    window.dt = new simpleDatatables.DataTable(dataTable, {
+        data,
+        perPage: 20,
+    });
+}
+
+function fileList(metadata) {
+    if (metadata) {
+        const list = (metadata.files || [])
+            .map(file => {
```

Comment:
Nit, but when I have very simple arrow functions like this i prefer to inline them:
`.map(file => file.path || "")`

Reply:
Funny given you wrote that :wink:. Will change in both places

Reply:
We could also create a common.js file potentially to hold these things... I may do that real fast